packages/hash/api/src/model/knowledge/orgMembership.model.ts

```diff
@@ -134,46 +146,43 @@ export default class extends EntityModel {
    * Get the user linked to the org membership.
    */
   async getUser(graphApi: GraphApi) {
-    const { data: incomingOrgMembershipLinks } = await graphApi.getLinksByQuery(
-      {
-        filter: {
-          all: [
-            {
-              equal: [{ path: ["target", "id"] }, { parameter: this.entityId }],
-            },
-            {
-              equal: [
-                { path: ["type", "versionedUri"] },
-                {
-                  parameter: SYSTEM_TYPES.linkType.hasMembership.schema.$id,
-                },
-              ],
-            },
-          ],
-        },
-        graphResolveDepths: {
-          dataTypeResolveDepth: 0,
-          propertyTypeResolveDepth: 0,
-          linkTypeResolveDepth: 0,
-          entityTypeResolveDepth: 0,
-          linkResolveDepth: 0,
-          linkTargetEntityResolveDepth: 0,
-        },
-      },
-    );
-
-    const [incomingOrgMembershipLinkRootedSubgraph] =
-      incomingOrgMembershipLinks;
-
-    if (!incomingOrgMembershipLinkRootedSubgraph) {
+    const incomingOrgMembershipLinkEntityModels =
+      await LinkEntityModel.getByQuery(graphApi, {
+        all: [
+          {
+            equal: [
+              { path: ["rightEntity", "uuid"] },
+              { parameter: this.entityUuid },
+            ],
+          },
+          {
+            equal: [
+              { path: ["type", "versionedUri"] },
+              {
+                parameter: SYSTEM_TYPES.linkEntityType.hasMembership.schema.$id,
+              },
+            ],
+          },
+          {
+            equal: [{ path: ["version"] }, { parameter: "latest" }],
+          },
+          {
+            equal: [{ path: ["archived"] }, { parameter: false }],
+          },
+        ],
+      });
+
+    const [incomingOrgMembershipLinkEntityModel] =
+      incomingOrgMembershipLinkEntityModels;
+
+    if (!incomingOrgMembershipLinkEntityModel) {
       throw new Error(
-        `Critical: org membership with entity id ${this.entityId} doesn't have a linked user`,
+        `Critical: org membership with entity id ${this.baseId} doesn't have a linked user`,
       );
     }
 
-    return await UserModel.getUserById(graphApi, {
-      entityId:
-        incomingOrgMembershipLinkRootedSubgraph.link.inner.sourceEntityId,
-    });
+    return UserModel.fromEntityModel(
+      incomingOrgMembershipLinkEntityModel.rightEntityModel,
```

Comment:
This should be `leftEntityModel` right?

Reply:
Good spot, fixed in https://github.com/hashintel/hash/pull/1435/commits/8b11f74135f8380398ab83c0e1a3005dd611be6b

Reply:
Any idea why this wouldn't have thrown any errors in tests or such? Do we want to improve testing to make sure the models are the types we expected?

Reply:
I did not fix the org memberships tests as part of this PR. I'm assuming we will fix all of them before merging `dev/links`, so we can spot things like this before merging.